platform/darwin/src/MGLPolygon.mm

```diff
@@ -155,6 +167,22 @@ - (NSUInteger)hash {
     return hash;
 }
 
+- (CLLocationCoordinate2D)coordinate {
+    CLLocationCoordinate2D centroid;
+    
+    mbgl::Polygon<double> geometry;
+    for (MGLPolygon *polygon in self.polygons) {
+        geometry.push_back(polygon.ring);
```

Comment:
This doesn’t account for holes in the polygon, which can affect the shape’s centroid. (We try to avoid labeling a polygon inside one of its holes.)

This class already has a `-geometryObject` method that returns a `mbgl::Geometry<double>`, but it actually creates an `mbgl::MultiPolygon<double>`. Consider factoring out a `-multiPolygon` method, similar to how `-[MGLPolygon geometryObject]` calls [`-[MGLPolygon polygon]`](https://github.com/mapbox/mapbox-gl-native/blob/4b84ae119dd9489b6fbd6a11eca2102e0a79747b/platform/darwin/src/MGLPolygon.mm#L69).